flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/SourceOperator.java

```diff
@@ -370,7 +413,14 @@ public DataInputStatus emitNext(DataOutput<OUT> output) throws Exception {
     private DataInputStatus emitNextNotReading(DataOutput<OUT> output) throws Exception {
         switch (operatingMode) {
             case OUTPUT_NOT_INITIALIZED:
-                currentMainOutput = eventTimeLogic.createMainOutput(output);
+                watermarkTrackingOutput = new WatermarkTrackingOutput<>(output);
+                if (watermarkAlignmentParams.isEnabled()) {
+                    processingTimeService.scheduleWithFixedDelay(
+                            this::emitLatestWatermark,
+                            watermarkAlignmentParams.getUpdateInterval(),
+                            watermarkAlignmentParams.getUpdateInterval());
+                }
+                currentMainOutput = eventTimeLogic.createMainOutput(watermarkTrackingOutput);
```

Comment:
thiis will most probably cause some tests to fail. 

We should use the original `output` if watermark alignment is disabled.